File: chiselbot_flutter/lib/models/inquiry.dart
enum InquiryStatus { WAITING, ANSWERED, CLOSED }

InquiryStatus statusFromString(String? s) {
  switch (s) {
    case 'WAITING':
      return InquiryStatus.WAITING;
    case 'ANSWERED':
      return InquiryStatus.ANSWERED;
    case 'CLOSED':
      return InquiryStatus.CLOSED;
    default:
      return InquiryStatus.WAITING;
  }
}

String statusToString(InquiryStatus st) {
  switch (st) {
    case InquiryStatus.WAITING:
      return 'WAITING';
    case InquiryStatus.ANSWERED:
      return 'ANSWERED';
    case InquiryStatus.CLOSED:
      return 'CLOSED';
  }
}

class Inquiry {
  final int inquiryId;
  final int? userId;
  final int? adminId;
  final String title;
  final String content;
  final String? answerContent;
  final InquiryStatus status;
  final DateTime createdAt;
  final DateTime? answeredAt;
  final DateTime? updatedAt;

  Inquiry({
    required this.inquiryId,
    required this.userId,
    required this.adminId,
    required this.title,
    required this.content,
    required this.answerContent,
    required this.status,
    required this.createdAt,
    required this.answeredAt,
    required this.updatedAt,
  });

  static DateTime? _parseDT(dynamic v) {
    if (v == null) return null;
    if (v is String) return DateTime.tryParse(v);
    // Timestamp를 직렬화해서 내려올 수 있으므로 문자열 변환 기대
    return DateTime.tryParse(v.toString());
  }

  factory Inquiry.fromJson(Map<String, dynamic> j) {
    // 키를 'id'로 보낼 수도, 'inquiryId'로 보낼 수도 있어 방어
    final id = j['inquiryId'] ?? j['id'];
    return Inquiry(
      inquiryId: (id is int) ? id : int.parse(id.toString()),
      userId: j['userId'] == null ? null : int.tryParse(j['userId'].toString()),
      adminId:
          j['adminId'] == null ? null : int.tryParse(j['adminId'].toString()),
      title: j['title'] ?? '',
      content: j['content'] ?? '',
      answerContent: j['answerContent'],
      status: statusFromString(j['status']?.toString()),
      createdAt: _parseDT(j['createdAt']) ?? DateTime.now(),
      answeredAt: _parseDT(j['answeredAt']),
      updatedAt: _parseDT(j['updatedAt']),
    );
  }
}

File: chiselbot_api_server/src/main/java/com/coach/chiselbot/domain/Inquiry/Inquiry.java
package com.coach.chiselbot.domain.Inquiry;

import com.coach.chiselbot.domain.user.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Inquiry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // 작성자
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    // 답변자(관리자) 임시 컬럼
    //private String adminName;

    // 답변자(관리자) - 답변 전엔 null 가능
    // admin 만들어지기 전에 연결 확인용
    @ManyToOne(fetch = FetchType.LAZY, optional = true)
    @JoinColumn(name = "admin_id")
    private User admin;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(columnDefinition = "TEXT")
    private String answerContent;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private InquiryStatus status = InquiryStatus.WAITING;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Timestamp createdAt;

    // 문의 등록 시간 임시 컬럼
    @Column(name = "answered_at") // nullable
    private Timestamp answeredAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Timestamp updatedAt;

}
